bin/index.js

```diff
@@ -1,288 +1,220 @@
+  ])
+  return response
 }
 
-function initializeSourceCode(outputDir) {
-    initializeIndexFile(outputDir);
-    initializeAppFile(outputDir);
-    initializeConfigFile(outputDir);
-    initializeModels(outputDir);
+async function generate (input) {
+  let outputDir = input['output-dir']
+  if (outputDir == null || outputDir.length === 0) {
+    outputDir = defaultOutputDir
+  }
+  let packageName = input['package-name']
+  if (packageName == null || packageName.length === 0) {
+    packageName = defaultPackageName
+  }
+
+  console.log('Initializing Tigris quickstart application')
+  initializeDirectoryStructure(outputDir)
+  initializePackageFile(outputDir, packageName)
+  initializeTSConfigFile(outputDir)
+  initializeSourceCode(outputDir)
+
+  console.log('🎉 Initialized the Tigris quickstart application successfully')
+  console.log('Run following command to install dependencies')
+  console.log('npm install')
+  console.log('Learn more at https://docs.tigrisdata.com/quickstart')
 }
 
-function initializeDirectoryStructure(outputDir) {
-    // create outputDir
-    if (!fs.existsSync(outputDir)) {
-        fs.mkdirSync(outputDir);
-    }
-    // create src directory
-    const srcDir = outputDir + '/src';
-    if (!fs.existsSync(srcDir)) {
-        fs.mkdirSync(srcDir);
-    }
-
-    const libDir = outputDir + '/src/lib';
-    if (!fs.existsSync(libDir)) {
-        fs.mkdirSync(libDir);
-    }
-
-    const modelsDir = outputDir + '/src/models';
-    if (!fs.existsSync(modelsDir)) {
-        fs.mkdirSync(modelsDir);
-    }
+function initializeSourceCode (outputDir) {
+  initializeModels(outputDir)
+  initializeCollectionsFile(outputDir)
+  initializeScriptFile(outputDir)
+  initializeTigrisClientFile(outputDir)
 }
 
-function initializePackageFile(outputDir, packageName) {
-    const content = '{\n' +
-        '  "name": "' + packageName + '",\n' +
-        '  "main": "index.js",\n' +
-        '  "scripts": {\n' +
-        '    "clean": "rm -rf dist",\n' +
-        '    "build": "npm run clean && npm install && npx tsc",\n' +
-        '    "start": "npm install && npm run build && node dist/index.js"\n' +
-        '  },\n' +
-        '  "dependencies": {\n' +
-        '    "@tigrisdata/core": "latest"\n' +
-        '  },\n' +
-        '  "devDependencies": {\n' +
-        '    "@types/node": "^17.0.42",\n' +
-        '    "typescript": "^4.7.3"\n' +
-        '  }\n' +
-        '}\n';
-    fs.writeFileSync(outputDir + '/package.json', content);
+function initializeDirectoryStructure (outputDir) {
+  // create outputDir
+  if (!fs.existsSync(outputDir)) {
+    fs.mkdirSync(outputDir)
+  }
+  // create src directory
+  const srcDir = outputDir + '/src'
+  if (!fs.existsSync(srcDir)) {
+    fs.mkdirSync(srcDir)
+  }
+
+  const modelsDir = outputDir + '/src/models'
+  if (!fs.existsSync(modelsDir)) {
+    fs.mkdirSync(modelsDir)
+  }
+
+  const collectionsDir = outputDir + '/src/collections'
```

Comment:
Is the convention to have a collections directory? Or would the code go somewhere else.

Reply:
probably repository